pkg/triggerconfig/merge/merge.go

```diff
@@ -0,0 +1,46 @@
+package merge
+
+import (
+	"github.com/jenkins-x/lighthouse/pkg/config"
+	"github.com/jenkins-x/lighthouse/pkg/config/job"
+	"github.com/jenkins-x/lighthouse/pkg/plugins"
+	"github.com/jenkins-x/lighthouse/pkg/triggerconfig"
+)
+
+// ConfigMerge merges the repository configuration into the global configuration
+func ConfigMerge(cfg *config.Config, pluginsCfg *plugins.Configuration, repoConfig *triggerconfig.Config, repoOwner string, repoName string) error {
+	repoKey := repoOwner + "/" + repoName
+	if len(repoConfig.Spec.Presubmits) > 0 {
+		if cfg.Presubmits == nil {
+			cfg.Presubmits = map[string][]job.Presubmit{}
+		}
+
+		var ps []job.Presubmit
+		for _, p := range repoConfig.Spec.Presubmits {
+			ps = append(ps, ToPresubmit(p))
+		}
+		cfg.Presubmits[repoKey] = ps
+	}
+	if len(repoConfig.Spec.Postsubmits) > 0 {
+		if cfg.Postsubmits == nil {
+			cfg.Postsubmits = map[string][]job.Postsubmit{}
+		}
+
+		var ps []job.Postsubmit
+		for _, p := range repoConfig.Spec.Postsubmits {
+			ps = append(ps, ToPostsubmit(p))
+		}
+		cfg.Postsubmits[repoKey] = ps
+	}
+
+	// lets make sure we've got a trigger added
+	idx := len(pluginsCfg.Triggers) - 1
```

Comment:
Do we really want to mutate this ?

Reply:
we want to add a trigger for each presubmit/postsubmit yet. Do you have a better suggestion?

Reply:
ah, it's not related to the trigger plugin configuration, i was probably mistaken. sorry.

Reply:
yeah - its just ensuring that a trigger exists for any presubmit/postsubmit defined in repo basically - its not configuring the trigger plugin